src/main/java/org/nixos/mvn2nix/URIBuilder.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2022 Benjamin Asbach
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+ * CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package org.nixos.mvn2nix;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+public final class URIBuilder {
+
+	private URIBuilder() {
+	}
+	
+	public static URI build(String base, URI baseSuffix) throws URISyntaxException {
```

Comment:
Comment this method please and lets make it not throw? Maybe Optional return?
I'm not a fan of checked exceptions in Java.

Reply:
Just to clarify my intentions: I wanted to extract the code which caused #10 in order to write a test and validate if the fix works without refactoring the original code.

I hate that `URISyntaxException` in general as well. On the other hand what do we win with an `Optional` here? We need more handling in this class and the handling in caller is as long as before.